plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/parsers/YamlPluginOutputConverter.java

```diff
@@ -0,0 +1,95 @@
+package io.jenkins.tools.pluginmanager.parsers;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import io.jenkins.tools.pluginmanager.impl.Plugin;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class YamlPluginOutputConverter implements PluginOutputConverter {
+    @Override
+    public String convert(List<Plugin> plugins) {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory().disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER))
+                .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        try {
+            return mapper.writeValueAsString(mapToOutputFormat(plugins));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return "Error converting to yaml";
+        }
+    }
+
+    private PluginHolder mapToOutputFormat(List<Plugin> plugins) {
+        List<PluginYaml> convertedPlugins = plugins.stream()
+                .map(this::toPluginHolder)
+                .collect(Collectors.toList());
+
+        return new PluginHolder(convertedPlugins);
+    }
+
+    private PluginYaml toPluginHolder(Plugin plugin) {
+        return new PluginYaml(plugin.getName(), plugin.getGroupId(), new PluginSource(plugin.getUrl(), plugin.getVersion().toString()));
+    }
+
+    public static class PluginHolder {
+        private final List<PluginYaml> plugins;
+
+        public PluginHolder(List<PluginYaml> plugins) {
+            this.plugins = plugins;
+        }
+
+        public List<PluginYaml> getPlugins() {
+            return plugins;
+        }
+    }
+
+    public static class PluginYaml {
```

Comment:
Please use the existing class? added in #114 ?

Reply:
🤣 🙄 oops

Reply:
Feel free to rename them though... As naming is hard!

Reply:
refactored now, PTAL